js/navigation.js

```diff
@@ -42,6 +42,45 @@ tabContainer.addEventListener('click', function (event) {
 
 var formElement = document.querySelector('.search-form');
 
+function recipeGenerator(recipe) {
+  var p = document.createElement('p');
+  p.onclick = function () {
+    window.open(recipe.url);
+  };
+  var pText = document.createTextNode(recipe.label);
+  p.appendChild(pText);
+
+  var favStar = document.createElement('i');
+
+  if (favorites.getRecipe(recipe.uri) !== undefined) {
+    favStar.className = 'fas fa-star favorite';
+  } else {
+    favStar.className = 'far fa-star favorite';
+  }
+  favStar.setAttribute('data-recipe-uri', recipe.uri);
+
+  var cartIcon = document.createElement('i');
+
+  if (shoppingList.getRecipe(recipe.uri) !== undefined) {
+    cartIcon.className = 'fas fa-shopping-cart shopping-cart';
+  } else {
+    cartIcon.className = 'fas fa-cart-plus shopping-cart';
+  }
```

Comment:
Both of these conditionals are searching for the same condition, they can be combined.